src/main/java/com/minecolonies/coremod/commands/GetColonyAndCitizen.java

```diff
@@ -0,0 +1,224 @@
+        {
+            try
+            {
+                colonyId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                colonyId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == SHORT_ARGUMENT_LENGTH || args.length == NAME_ARGUMENT_LENGTH)
+        {
+            if (ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID) == null)
+            {
+                colonyId = NO_COLONY_CODE;
+            }
+            else
+            {
+                colonyId = ColonyManager.getIColonyByOwner(sender.getEntityWorld(), mayorID).getID();
+            }
+        }
+        else if (args.length >= TOO_MANY_ARGUMENTS_LENGTH)
+        {
+            colonyId = TOO_MANY_ARGUMENTS_CODE;
+        }
+        else
+        {
+            colonyId = UNKNOWN_ERROR_CODE;
+        }
+        if (colonyId >= 0)
+        {
+            if (ColonyManager.getColony(colonyId) == null)
+            {
+                colonyId = NOT_FOUND_CODE;
+            }
+        }
+
+
+        return colonyId;
+    }
+
+    /**
+     * Getting the citizen ID.
+     * @param colonyId          The colony ID for getting the citizen ID
+     * @param args              The given arguments
+     * @return Returns citizen ID
+     */
+    public static int getCitizenId(@NotNull final int colonyId, @NotNull final String... args)
+    {
+        int citizenId;
+        String citizenName;
+        citizenId = NOT_FOUND_CODE;
+        if (args.length == NORMAL_ARGUMENT_LENGTH)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[1]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == SHORT_ARGUMENT_LENGTH)
+        {
+            try
+            {
+                citizenId = Integer.parseInt(args[0]);
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = ONLY_NUMBERS_CODE;
+            }
+        }
+        else if (args.length == NAME_ARGUMENT_LENGTH && colonyId >= 0)
+        {
+            citizenName = args[0] + " " + args[1] + " " + args[2];
+            for (int i = 1
+                   ; i <= ColonyManager.getColony(colonyId).getCitizens().size(); i++)
+            {
+                if (ColonyManager.getColony(colonyId).getCitizen(i).getName() != null)
+                {
+                    if (ColonyManager.getColony(colonyId).getCitizen(i).getName().equals(citizenName))
+                    {
+                        citizenId = i;
+                    }
+                }
+            }
+
+        }
+        else if (args.length == ID_AND_NAME_ARGUMENT_LENGTH && colonyId >= 0)
+        {
+            citizenName = args[1] + " " + args[2] + " " + args[3];
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)